src/librbd/exclusive_lock/PostAcquireRequest.cc

```diff
@@ -114,7 +115,7 @@ void PostAcquireRequest<I>::send_open_journal() {
   }
   if (!journal_enabled) {
     apply();
-    finish();
+    send_open_image_cache();
```

Comment:
Nit: this is being executed after the "successful" `apply` state

Reply:
@dillaman I am confused about this comment, could you give more explanation?

Reply:
The `apply` method is executed on line 117 and then you continue to open the image cache (which could fail).

Reply:
This part makes me confused.   
Please have a look at https://github.com/ceph/ceph/blob/master/src/librbd/exclusive_lock/PostAcquireRequest.cc#L130 
It calls apply() before m_journal->open(ctx).  Based on my understanding, the function apply() should be called when all the steps in PostAccquireRequest are completed.   
So why does it call apply() before opening the journal? It also could fail.

Reply:
I added logic to shutdown the cache when the initialization fails.

Reply:
It's because journal replay might need to update the object-map. If it hits a failure, it will revert the `apply` state. I guess technically the RWL cache would also need the object-map if it can restart the writeback immediately after initialization. You would still potentially need to `revert` it, though.